sstables/sstable_set.cc

```diff
@@ -342,6 +347,30 @@ seastar::shared_ptr<sstable_set_impl> partitioned_sstable_set::clone() const {
     return seastar::make_shared<partitioned_sstable_set>(_schema, _unleveled_sstables, _leveled_sstables, _all, _all_runs, _use_level_metadata, _bytes_on_disk);
 }
 
+future<seastar::shared_ptr<sstable_set_impl>> partitioned_sstable_set::clone_gently() const {
+    std::vector<shared_sstable> unleveled_sstables;
+    interval_map_type leveled_sstables;
+    lw_shared_ptr<sstable_list> all = _all;
+    std::unordered_map<run_id, sstable_run> all_runs;
+
+    // We presume that copying a vector<shared_sstable> won't stall
+    unleveled_sstables = _unleveled_sstables;
+
+    for (const auto& [interval, set] : _leveled_sstables) {
+        co_await coroutine::maybe_yield();
+        // We presume that copying a single unordered_set<shared_sstable> in each level won't stall
+        leveled_sstables.add({interval, set});
+    }
+
+    all_runs.reserve(_all_runs.size());
+    for (const auto& [run_id, run] : _all_runs) {
+        co_await coroutine::maybe_yield();
+        all_runs.emplace(run_id, run);
+    }
+
+    co_return seastar::make_shared<partitioned_sstable_set>(_schema, std::move(unleveled_sstables), std::move(leveled_sstables), std::move(all), std::move(all_runs), _use_level_metadata, _bytes_on_disk);
+}
```

Comment:
partitioned_sstable_set needs another ctor. the existing one takes leveled_sstables, etc, as const ref, meaning copy won't be avoided.